src/plugins/data/public/search/expressions/esaggs.ts

```diff
@@ -65,12 +75,15 @@ interface Arguments {
   partialRows: boolean;
   includeFormatHints: boolean;
   aggConfigs: string;
+  timeField?: string[];
```

Comment:
remove ?

Reply:
The parameter is not defined is required in the definition below, so it can be skipped and thus can be `undefined`. Actually the typing of the other parameters here is wrong, because they should also all have `?` or being marked as `required: true` in their definition, but I didn't want to fix this in this PR.

Reply:
so we have timeField and timeFields ?

Reply:
> Actually the typing of the other parameters here is wrong, because they should also all have `?` or being marked as `required: true` in their definition, but I didn't want to fix this in this PR.

@timroes This actually highlights weird thing about the Expression function typings which I don't think we have been doing consistently:

The types are what get enforced inside the function body. So while any arg with `default: 'whatever'` set is technically optional for _users_ of the expression function, once you are _inside_ the expression function itself, if it is not marked with `?` in the arguments interface, TS will warn that the arg can be possibly `undefined` (since TS doesn't know that the defaults get injected at runtime).

So which is right? Do we write the typings to be accurate for the users of the functions (which means lots of `?`), at the expense of function authors needing a lot of unnecessary `if (args.whatever)` checks, or `args.whatever!` ...  Or do we write the typings to be accurate for the function authors, at the expense of users of the functions not having accurate typings?

IMO your proposed approach is probably better (include `?` if there's a `default` or no `required: true`). It just means needing to update many of our function definitions (including Canvas functions which seem to have this discrepancy as well).

Reply:
^ Thinking about this more, in `ExpressionFunctionDefinition` we might be able to enforce the presence of `required: true` for non-optional args.

At any rate; that's a topic for a different thread 🙂